fbw-common/src/wasm/systems/systems/src/air_conditioning/mod.rs

```diff
@@ -63,34 +60,109 @@ pub trait AdirsToAirCondInterface {
+    fn hot_air_pushbutton_is_on(&self) -> bool;
+    fn cabin_fans_is_on(&self) -> bool;
+    fn flow_selector_position(&self) -> OverheadFlowSelector;
+}
+
+pub trait PressurizationOverheadShared {
+    fn is_in_man_mode(&self) -> bool;
+    fn ditching_is_on(&self) -> bool;
+    fn ldg_elev_is_auto(&self) -> bool;
+    fn ldg_elev_knob_value(&self) -> f64;
+}
+
+/// Cabin Zones with double digit IDs are specific to the A380
+/// 1X is main deck, 2X is upper deck
 pub enum ZoneType {
     Cockpit,
     Cabin(u8),
+    Cargo(u8),
 }
 
 impl ZoneType {
     fn id(&self) -> usize {
         match self {
             ZoneType::Cockpit => 0,
-            ZoneType::Cabin(number) => *number as usize,
+            ZoneType::Cabin(number) => {
+                if number < &10 {
+                    *number as usize
+                } else if number < &20 {
+                    *number as usize - 10
+                } else {
+                    *number as usize - 13
+                }
+            }
+            ZoneType::Cargo(number) => *number as usize + 15,
         }
     }
 }
 
-// TODO: At the moment this lives here but it's specific to the A320.
 impl Display for ZoneType {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         match self {
             ZoneType::Cockpit => write!(f, "CKPT"),
             ZoneType::Cabin(number) => match number {
                 1 => write!(f, "FWD"),
                 2 => write!(f, "AFT"),
-                _ => panic!("Not implemented for the A320 aircraft."),
+                11 => write!(f, "MAIN_DECK_1"),
+                12 => write!(f, "MAIN_DECK_2"),
+                13 => write!(f, "MAIN_DECK_3"),
+                14 => write!(f, "MAIN_DECK_4"),
+                15 => write!(f, "MAIN_DECK_5"),
+                16 => write!(f, "MAIN_DECK_6"),
+                17 => write!(f, "MAIN_DECK_7"),
+                18 => write!(f, "MAIN_DECK_8"),
+                21 => write!(f, "UPPER_DECK_1"),
+                22 => write!(f, "UPPER_DECK_2"),
+                23 => write!(f, "UPPER_DECK_3"),
+                24 => write!(f, "UPPER_DECK_4"),
+                25 => write!(f, "UPPER_DECK_5"),
+                26 => write!(f, "UPPER_DECK_6"),
+                27 => write!(f, "UPPER_DECK_7"),
+                _ => panic!("Not implemented for this aircraft."),
             },
+            ZoneType::Cargo(number) => match number {
+                1 => write!(f, "CARGO_FWD"),
+                2 => write!(f, "CARGO_BULK"),
+                _ => panic!("Not implemented for this aircraft."),
+            },
+        }
+    }
+}
+
+#[derive(Clone, Copy)]
+pub enum OverheadFlowSelector {
+    Lo = 80,
+    Norm = 100,
+    Hi = 120,
+    Man = 0,
+}
+
+read_write_enum!(OverheadFlowSelector);
+
+impl From<f64> for OverheadFlowSelector {
+    fn from(value: f64) -> Self {
+        match value as u8 {
+            // Note: A380 overhead only has 0,1,2 positions, it doesn't rotate to MAN
```

Comment:
I think it's the 320 that doesn't have the MAN knob or am I misunderstanding this?

Reply:
Good spot! To be honest I'm not happy with how this is set up at the moment because it doesn't allow for the different positions of the A380. Just pushed a commit to fix it 😄.

Reply:
For now I left the a380 reading the selector as if it was the a320 until the underlying system is modelled.

Reply:
Great, splitting them up is a good idea. However, at this point, it's probably beneficial to make two separate implementations of the `OverheadFlowSelector` as we don't really want to have a distinction (such as `from_a320` and `from_a380` functions) between the planes in the shared code. I realized that this might need a bunch of follow-up changes, so do let me know if you want to leave that change for a later PR or if you think it's not a good idea at all 😄.

Reply:
Following our conversation on Discord, I've now removed the `from` trait altogether. It was only being isued within the `read` functions, so I moved the code there directly.